packages/app/src/server/service/passport.ts

```diff
@@ -719,8 +722,40 @@ class PassportService implements S2sMessageHandlable {
       return response.statusCode === 200;
     }
     catch (err) {
-      logger.debug('Issuer host unreachable:', err.code);
+      logger.error('OidcStrategy: issuer host unreachable:', err.code);
+    }
+  }
+
+  /**
+   * Get oidcIssuer object
+   * Utilize p-retry package to retry oidcIssuer initialization 3 times
+   *
+   * @param issuerHost
+   * @returns instance of OIDCIssuer
+   */
+  async getOIDCIssuerInstace(issuerHost) {
+    const oidcIssuerHostReady = await this.isOidcHostReachable(issuerHost);
+    if (!oidcIssuerHostReady) {
+      logger.error('OidcStrategy: setup failed: OIDC Issur host unreachable');
+      return;
     }
+    const oidcIssuer = await pRetry(async() => {
+      return OIDCIssuer.discover(issuerHost);
+    }, {
+      onFailedAttempt: (error) => {
+        if (error.attemptNumber === 1) {
+          logger.debug('OidcStrategy: setup failed. Retrying ...');
+        }
+        else {
+          logger.debug(`OidcStrategy: setup attempt ${error.attemptNumber - 1} failed. Retrying ...`);
```

Comment:
I think you don't have to minus 1 from attemptNumber. What do you think? You can decide if you have a specific reason.

Reply:
I used the p-retry library with the retry option: 3, but during testing there were 4 tries, so I modify the log. I will set the attemptNumber with the original one along with error logging.  In your opinion, What is the most reasonable retry to set?